packages/date/src/index.js

```diff
@@ -580,7 +580,7 @@ function buildMoment( dateValue, timezone = '' ) {
 	const dateMoment = momentLib( dateValue );
 
 	if ( timezone && ! isUTCOffset( timezone ) ) {
-		return dateMoment.tz( String( timezone ) );
+		return dateMoment.tz( /** @type {string} */ ( timezone ) );
```

Comment:
We could add a comment here explaining that the `! isUTCOffset()` check guarantees that `timezone` is not a `number`, and therefore is a `string`